.buildkite/pipeline_pr_optional.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Generate Buildkite pipelines dynamically"""
+
+from common import COMMON_PARSER, group, pipeline_to_json
+
+# Buildkite default job priority is 0. Setting this to 1 prioritizes PRs over
+# scheduled jobs and other batch jobs.
+DEFAULT_PRIORITY = 1
+
+args = COMMON_PARSER.parse_args()
+
+defaults = {
+    "instances": args.instances,
+    "platforms": args.platforms,
+    # buildkite step parameters
+    "timeout_in_minutes": 45,
+    "env": dict(args.step_env),
+    # some optional tests are performance, so make sure they get ag=1 instances
+    "priority": DEFAULT_PRIORITY + 1,
+    "agent_tags": ["ag=1"],
+}
+defaults.update(args.step_param)
+
+optional_grp = group(
+    "❓ Optional",
+    # We need to pass both --optional and -m optional here: The first causes our custom collection hook to
+    # not skip the optional tests, while the second causes pytest to select all non-optional tests.
+    "./tools/devtool -y test -- ../tests/integration_tests/ --optional -m optional",
```

Comment:
I would add here flags for cpusets and numa nodes, similar to `pipeline_perf.py`: `-c 1-10 -m 0` to minimize variability in the tests.

Reply:
done! I tried limiting the cpu range more, too, but it caused hangs. I suppose it makes sense because the benchmarking process isnt the only affected by this, but rather the entire docker container